meetings/2024/LDM-2024-04-24.md

```diff
@@ -0,0 +1,86 @@
+# C# Language Design Meeting for Apr 24, 2024
+
+## Agenda
+
+- [Adjust dynamic binding rules for a situation of a single applicable candidate.](#adjust-dynamic-binding-rules-for-a-situation-of-a-single-applicable-candidate)
+
+## Quotes of the day
+
+"Once people are in dynamic-land they've already chosen"
+
+## Adjust dynamic binding rules for a situation of a single applicable candidate
+
+(https://github.com/dotnet/csharplang/pull/8027)
+
+The params collection work includes some changes to how binding occurs if either the primary expression or an argument is dynamic. When there is only one candidate, it could be used.
+
+This change caused [unexpected breaking changes in certain cases](https://github.com/dotnet/roslyn/issues/72750):
+
+> ```csharp
+> using System.Text.Json;
+> 
+> public class C
+> {
+>     public static C M(IFoo foo, dynamic value)m
+>     {
+>         var result = foo.Bar("name", value);
+>         return JsonSerializer.Deserialize<C>(result);
+>     }
+> }
+> 
+> public interface IFoo
+> {
+>     object Bar(string name, object value);
+> }
+> ```
+> 
+> After the recent changes around dynamic, the type of result is now object whereas previously it was dynamic.
+
+
+### Discussion
+
+It is desirable to use use static binding to make more things possible with dynamics. This can be done when there is only one applicable candidate.
+
+#### Dynamic or static result
+
+We do not plan to change things that work today, such as local functions. We are only considering whether new cases should have a static result or be "dynamified"
+
+There were concerns about returning static results from operations with dynamic being surprising. Some current cases, such as constructors do this - but that is because you are stating the return type explicitly in the call to `new`.
```

Comment:
Slight rewording to make the prose read a bit more fluently (to me, at least). The given reason only applies to constructors, not any other examples.

```suggestion
There were concerns about returning static results from operations with dynamic being surprising. Constuctors do this, but that is because you are stating the return type explicitly in the call to `new`.
```